### ISSUE TITLE
abandoned?

### ISSUE THREAD

**Issue by Steve-CCity** [moezbhatti/qksms#1970]
I'm wondering if anyone is working on keeping this app working.
Seems like maybe it's been abandoned, which would be sad, it's such a great message tool.

If so, thoughts on alternatives?

**Comment by Chloe-H** [comment]
Wondering the same thing. Literally looked up [this](https://alternativeto.net/software/qksms/) today. I'd really rather not leave QKSMS behind, though...it's been a good app. I just want support for MMS backups/transfers and I'll be happy!

**Comment by Chloe-H** [comment]
Quick update: I've been dipping my toes into [Textra](https://alternativeto.net/software/textra/about/), and I like it a lot. Scheduled messages, pinned conversations, custom conversation names, highly customizable colors and behavior. It even has the per-contact color customization options that QKSMS has.
Between that and using [Synctech's SMS Backup & Restore](https://www.synctech.com.au/sms-backup-restore/) app to transfer SMS _and_ MMS messages from my previous phone, I'm pretty happy.

The only other texting app I was even considering from the article I linked was [Mood Messenger](https://alternativeto.net/software/mood-sms/about/), since it is also listed as having a lot of features I care about _and_ seems to be privacy-focused.

![image](https://github.com/moezbhatti/qksms/assets/10136444/2a232de0-f68c-4412-b9aa-b070f6942570)

**Comment by Steve-CCity** [comment]
Thanks @Chloe-H 
I tried both Textra & Mood and went back to QKSMS.  Neither of them worked better for me.  My issues with QKSMS are pretty minor.  It works reliably for me.  I would just like to be able to Name Group Convos and Customize Group Notifications. And, of course wish that it was not abandoned   :  )

**Comment by 87113** [comment]
Hi, just to say that both are not opensource so for me it's not an option, but thank's for sharing.

I hope that qksms is not dead ;)

**Comment by random-pixels** [comment]
Just got an updated phone and was really disappointed by the new stock SMS apps; finally settled on QKSMS only to find it seems abandoned - no releases for 2.5 years now :(

UPDATE: There are newer versions on F-Droid. A [version](https://f-droid.org/en/packages/com.moez.QKSMS/) was added in January which is somehow three times the size of the previous release. 

Apparently it's only a [pre-release version](https://github.com/moezbhatti/qksms/releases), but.. maybe there is hope that a new version will eventually come out? :crossed_fingers: 

I installed the updated pre-release version and it seems to work well (Android 13). So for those of you willing to install the .apk rather than go through the Play Store, this should work for you and at least it's less than a year old. Sadly it does not have the features I want and I am not hopeful that they will be added if I submit a feature request, so I won't be keeping it.

@Chloe-H Thanks for the recommendation for Textra! I looked into it and it seems like a good alternative, albeit closed source. At least the paid option is a one-time fee and not a subscription (at least as of a few years ago based on the reviews I could find). Please correct me if I am wrong.

Questions: does Textra allow to configure timestamps to display on every single message (Update: found answer: no), and also, does it allow to configure a sound notification for a new message even when the app is open (Update: found answer: yes)? Most times I set the phone down after sending a text and don't get notified of the reply since the app is still open. These are the only 2 features missing from the the stock apps that I could not determine if Textra has or not. 

It seems documentation for SMS apps in general is quite lacking.

@87113 You can look into Partisan SMS, it is a fork of QKSMS and is open source that uses encryption. I don't know if it's on the Play Store. If not, you can try the pre-release 3.10.1 version I linked to above.

**Comment by Steve-CCity** [comment]
After trying Textra and Mood I went back to QKSMS then noticed what @random-pixels said about the latest Version - 3.10.1 being Pre- release, then I remembered one time loading up a prior version just to be able to name my Group Convos and then updating again.  Realized now that maybe the older version - 3.94 might be the one to use - since it worked fine.  So - I reverted back and boom - I can both Name Group Convos and Customize Group Notifications.  I'm thinking that these problems with the newer PreRelease versions are why they are still PreRelease - duh.

**Comment by Skilly146** [comment]
I have also been using the second most recent version 3.9.4 and haven't had any issues yet. Obviously won't work forever, but hopefully it will work until I can find an alternative or this project gets restarted

**Comment by 002gRTEah3mRYH** [comment]
She should fork.. that's a good app but MMS not working for me

**Comment by adamhotep** [comment]
See also #1964 (which has no responses), in which the poster claims the author is not responding to email either.

**Comment by tcecyk** [comment]
@random-pixels 

> which is somehow three times the size of the previous release.

was curious, that is in part due to the bundled librealm-jni.so getting bigger. realm.io is the on-device database used inside qksms

**Comment by mddvul22** [comment]
I've been using QKSMS for about 4 years.  Development has definitely slowed.  In the last year bugs have started creeping in to the application.  For example, if I go into my gallery and select an image and share it via QKSMS, I can't select who to send the image to.  In the place where I would select the person, it shows //media/external/imag... The only way that I can send an image via SMS is to start the text message to the person, and then click on the + symbol and choose the option to attach a photo.  Going the other way no longer works for me.

Nevertheless, I'm afraid that there won't be much more development work on QKSMS.  SMS is dying, probably to be replacing with either RCS or MLS.  Hopefully that future will include some open source applications.  But, there really needs to be some standardization.  The future looks dim for texting.  I don't want a future where I have to have Facebook Messenger, Google Messages, WhatsApp, Signal and a broken SMS app all installed to be able to communicate with everybody.

I'm a Lineage for Microg user.  If I have to move to Google Messages in the future, I'll have to move to regular Lineage with Gapps, which seems awful to me.  The problems with QKSMS are such that a few months ago I moved to the AOSP messaging app.  I used it for several weeks and then I just stopped receiving SMS messages with it.  That was a major failure and I moved back to QKSMS.  At this point, I suppose I'm stuck using QKSMS and will limp along with it until the future texting landscape settles into some kind of order.

Personally, I don't mind @moezbhatti moving  on.  However, I do wish that they would at least comment on what is going on, and/or offer to hand over maintenance to somebody else.

**Comment by Steve-CCity** [comment]
Thanks for your thoughtful comment @mddvul22 .  I learned a lot of things that I did not know before from reading it.

**Comment by fallsdevil** [comment]
I saw this app on a foss list, apparently this app has a 'replacement' I just downloaded this one [Partisan SMS](https://github.com/wrwrabbit/Partisan-SMS/releases/tag/v4.0.0) and it has a recent update from May, and in the app in the settings the same name appears **QKSMS** I found this interesting and I saw that some of you are looking for alternatives so I decided to make this comment...

**Comment by random-pixels** [comment]
> I saw this app on a foss list, apparently this app has a 'replacement' I just downloaded this one [Partisan SMS](https://github.com/wrwrabbit/Partisan-SMS/releases/tag/v4.0.0) and it has a recent update from May, and in the app in the settings the same name appears **QKSMS** I found this interesting and I saw that some of you are looking for alternatives so I decided to make this comment...

Yep, as I mentioned above, Partisan is a fork of QKSMS that uses encryption. Good to know it's still maintained.

**Comment by Dyras** [comment]
Sadly, the Partisan SMS fork also seems to be abandoned. See: https://github.com/wrwrabbit/Partisan-SMS/issues/34

**Comment by mddvul22** [comment]
If you want an open source, currently supported, SMS app, I've moved to Simple Mobile Tools SMS Messenger.  It is on [F-Droid](https://f-droid.org/en/packages/com.simplemobiletools.smsmessenger/) and in the [Google Play store](https://play.google.com/store/apps/details?id=com.simplemobiletools.smsmessenger&hl=en_US&gl=US).

**Comment by KaKi87** [comment]
@octoshrimpy What about [your fork](https://github.com/octoshrimpy/quik) ? Are you still planning on maintaining it ? If yes, would you enable issues on it ? Thanks

**Comment by 87113** [comment]
This is another fork of qksms : https://gitlab.e.foundation/e/os/Message looks like it's maintained by the eOS team.

**Comment by KaKi87** [comment]
Would they accept feature requests though ? 🤔

**Comment by 87113** [comment]
I have no idea. 
I can say that I use eOS and this app 'Message' on my phone and it's working for me. :)

**Comment by KaKi87** [comment]
How did you download it ? I can't even find an APK. 😅

**Comment by 87113** [comment]
> How did you download it ? I can't even find an APK. 😅

Ho! 🧐🧐🧐

It's installed by default in eOS. 
Maybe they don't provide public apk!?
I don't know.

**Comment by KaKi87** [comment]
Well then most of QKSMS users, who aren't using that ROM, can't install it, therefore it's not an actionable fork.

**Comment by 87113** [comment]
Hi, actually you can find the apk there : https://gitlab.e.foundation/e/os/Message/-/artifacts

**Comment by adamhotep** [comment]
> Hi, actually you can find the apk there : https://gitlab.e.foundation/e/os/Message/-/artifacts

Yes, but if you install it directly, you won't get security updates and bugfixes. That's why it would be ideally included in F-Droid and/or Google Play (in addition to whatever /e/OS uses).

**Comment by KaKi87** [comment]
Actually, even just uploading APKs to the git *releases* page would be fine thanks to [Obtainium](https://github.com/ImranR98/Obtainium), but not only the CI dowloads can't be automated but the APK inside these ZIPs can't even be installed because of being unsigned.

**Comment by alekksander** [comment]
just tried that efoundation fork. nothing seems different comparing to qksms. what's the point?

**Comment by Dyras** [comment]
@alekksander 
The point is that if there is ever a security hole in QKSMS it's never getting fixed.

Also, the API level will never increase so QKSMS will never have new security features.

**Comment by adamhotep** [comment]
> Actually, even just uploading APKs to the git _releases_ page would be fine thanks to [Obtainium](https://github.com/ImranR98/Obtainium), but not only the CI dowloads can't be automated but the APK inside these ZIPs can't even be installed because of being unsigned.

Obtainium is neat, but the /e/OS Messages app doesn't have any [releases](https://gitlab.e.foundation/e/os/Message/-/releases) and (as you note) the [artifacts](https://gitlab.e.foundation/e/os/Message/-/artifacts) are archives with unsigned APKs.

**Comment by octoshrimpy** [comment]
> @octoshrimpy What about [your fork](https://github.com/octoshrimpy/quik) ? Are you still planning on maintaining it ? If yes, would you enable issues on it ? Thanks

I would be down for it! I forked for the heck of it, in case, but I'm def down to help keep a maintained fork, but am gonna need help :)

opening up issues and PRs

**Comment by Dyras** [comment]
@octoshrimpy 
Any chance you could fork the eOS Messages app and just make that easily downloadable on non-eOS phones? I think that would be the easiest way of keeping the project alive without too much maintaining from your side. I'd happily be a beta tester if needed.

**Comment by octoshrimpy** [comment]
> @octoshrimpy Any chance you could fork the eOS Messages app and just make that easily downloadable on non-eOS phones? I think that would be the easiest way of keeping the project alive without too much maintaining from your side. I'd happily be a beta tester if needed.

if you'd like to fork eOs Messages and merge into Quik, i'll accept PRs and start learning how to compile it all.

**Comment by Dyras** [comment]
@octoshrimpy I sadly lack the skills of forking an Android app, but I'd happily help out with testing etc. I'm probably not alone. Maybe ask around at Privacy Guides, etc and see if someone else is up to the challenge?

Heck, maybe contact eOS and link them here and tell them there's interest in using their app?

**Comment by Joeviocoe** [comment]
Looks to be abandoned.  Recommending users uninstall and use something else.

**Comment by upbox-org** [comment]
For the adventurous: there is also "Deku SMS" available at F-Droid.
https://www.f-droid.org/en/packages/com.afkanerd.deku/
https://github.com/deku-messaging/Deku-SMS-Android
It's pretty new and focused on encryption. The devs describe it as "feature-rich" but I think it's minimalistic.

I'll also go with:
https://www.f-droid.org/en/packages/com.simplemobiletools.smsmessenger/
Got almost all features I need and is developed by a well known dev who is active for years in the OSS community.

**Comment by alekksander** [comment]
> For the adventurous: there is also "Deku SMS" available at F-Droid.

sadly it lacks nearly everything and currently the only feature is encryption. not a substitute to qksms, at least not yet.

> For the adventurous: there is also "Deku SMS" available at F-Droid. https://www.f-droid.org/en/packages/com.afkanerd.deku/ https://github.com/deku-messaging/Deku-SMS-Android It's pretty new and focused on encryption. The devs describe it as "feature-rich" but I think it's minimalistic.
> 
> I'll also go with: https://www.f-droid.org/en/packages/com.simplemobiletools.smsmessenger/ Got almost all features I need and is developed by a well known dev who is active for years in the OSS community.

this one is good, but still misses few features to actually compete with qksms. development is ongoing, but slow. my suspicion is that dev is focusing on other apps rather than messaging one.

**Comment by octoshrimpy** [comment]
I managed to compile my fork [quik](https://github.com/octoshrimpy/quik) last night, after merging a few PRs for bugfixes (like the spaces in search freeze app)

I have not managed to get github actions to build the APK, but it does compile locally. That's next on my to-learn list.

**Comment by octoshrimpy** [comment]
Quik is now up to date, compiling proper releases, as well as has a few bug fixes. spaces in search no longer crash, you can delete scheduled messages, all QK+ features enabled, and the next milestone is a revamp over scheduled messages, so you can see them inside the conversation before they send. 

I'm very new to android dev, so any help would be appreciated :)

**Comment by KaKi87** [comment]
Could you please sign the APKs ? Thanks

**Comment by octoshrimpy** [comment]
> Could you please sign the APKs ? Thanks

First time doing this, but I think I got it working.

https://github.com/octoshrimpy/quik/releases/tag/v4.0.5

let me know?

**Comment by KaKi87** [comment]
Looks good !

**Comment by Joeviocoe** [comment]
> I managed to compile my fork [quik](https://github.com/octoshrimpy/quik) last night, after merging a few PRs for bugfixes (like the spaces in search freeze app)
> 
> I have not managed to get github actions to build the APK, but it does compile locally. That's next on my to-learn list.

Since this is a fork, we should still consider the moezbhatti/qksms abandoned and move discussion of the octoshrimpy fork to its own github project.
@Steve-CCity  can close this issue.

**Comment by alekksander** [comment]
@octoshrimpy 
Thanks for the fork, looks promising.
If You plan to add message encryption, then partisan is probably worthy looking at.
Also please consider implementing „material you” for a fresh visual start.

**Comment by mddvul22** [comment]
Earlier in this thread, I mentioned that I had moved to Simple Mobile Tools SMS Messenger, which was being actively developed and was in F-Droid.  However, the useful future of that app is seriously in doubt.  The developer of that app has just sold it to a company known for putting spyware and adware into their apps.  If you want to read a humorous/interesting thread, see here:

https://github.com/SimpleMobileTools/General-Discussion/issues/241

It looks like another one of the developers of that app has now forked it.  The future of open source SMS apps continues to be very unstable.

**Comment by Dyras** [comment]
Here's a link to the SMT forks mentioned in the post above:
https://github.com/FossifyOrg

**Comment by KaKi87** [comment]
Wow, they moved fast. Nobody even knows whether the buyer will actually do what people fear they would do.

**Comment by Dyras** [comment]
Well the developer pulling this sale out of the blue ruined any and all cred he had in the FOSS industry in many eyes. It's only natural that we see an unofficial continuation even if he backs out of the sale or something similar.

**Comment by KaKi87** [comment]
Actually I assumed that it would go through, what I meant is nobody knows whether the future owner will actually add ads and/or malware into the apps.

**Comment by RunningDroid** [comment]
> nobody knows whether the future owner will actually add ads

They're already saying they've added ads: https://play.google.com/store/apps/details?id=com.simplemobiletools.smsmessenger
![Screenshot 2023-12-06 at 23-27-37 Simple SMS Messenger - Apps on Google Play](https://github.com/moezbhatti/qksms/assets/486393/28668f05-b48a-4235-8846-54d3ee1f0346)

**Comment by KaKi87** [comment]
Mea culpa.

I see "added trial" in changelogs and "pro" versions as well. :/

Fossify actually isn't moving fast enough then, because users are getting those updates and can't switch as it seems these forks aren't distributed anywhere yet.

**Comment by alekksander** [comment]
@octoshrimpy is there anything going on regarding quik?

**Comment by octoshrimpy** [comment]
> @octoshrimpy is there anything going on regarding quik?

Im still learning kotlin and trying to figure out why obtanium won't grab the latest version. Otherwise i am working on submitted bugs first. :)

**Comment by gcb** [comment]
lots of info here. Anyone actually uses SMS in this day and age for anything more than (unsafe) two factor auth?

Sending images and group chats were never SMS, but MMS (basically, send a url in SMS and download that url and show in the client instead. I usually block it --edit: just realized there no more option for it. must change on the main xml file to disable it now in qksms. too much trouble).

google et al are all replacing SMS with RCS. which is MMS done just a little less worse. I.e. it uses https and notifications all the way, no more SMS backend (plus the much worse entire-contact-list-sent-to-server-all-the-time) This is mostly being done because telephony and SMS have privacy laws in most countries and RCS will not ;) it works same as email. hurray for advertising and citizen spying.

### anyway, here are some summary notes from the discussion here so far:

- updates on this repo: moving from  3.9 to 3.10RC, huge increase in size, here are the diffs https://github.com/moezbhatti/qksms/compare/v3.9.4...v3.10.0
- e fork: I can mostly see translation and icon changes. They did remove some unused code and things like "rate qksms" dialogs. https://gitlab.e.foundation/e/os/Message/-/commits/main/?ref_type=HEADS
- octoshrimpy fork: really hard to follow. Unsure if forked from 3.9 or 3.10, but now it also have a 3.10 now (same?), not sure if imported changes from e? most commits seem to be renames and build related https://github.com/moezbhatti/qksms/compare/master...octoshrimpy:quik:master
- Partisan: just added encryption on top of qksms, from a user called [vivabelarus](https://github.com/vivabelarus) :/ unsure how to read nationalism for belarus but anyway. it was then further forked by someone who seems to hate the current president making it more confusing. They seem to have forked 3.9 and not touched the codebase besides the encryption stuff https://github.com/wrwrabbit/Partisan-SMS 
- more forks?

alternatives mentioned:
- https://github.com/deku-messaging/Deku-SMS-Android also abandoned. but it is mostly a prototype to sending SMS from the phone who in turn receives/sends the contents via other channels (highly recommend kdeconnect for that instead)
- https://github.com/SimpleMobileTools/Simple-SMS-Messenger seems OK gpl3. updated. also handles MMS and number blocking.
- some closed source alternatives I will ignore.
- more alternatives?

I'm thinking we all should jump ship and start contributing to the simpleMobileTools stuff.

**Comment by upbox-org** [comment]
> I'm thinking we all should jump ship and start contributing to the simpleMobileTools stuff.

It seems you missed that the SimpleMobileTools dev sold his project recently to a company known to buy free projects, close them and monetize them in an excessively manner.

Take a look at the fork instead and consider to contribute there:
https://github.com/FossifyOrg

**Comment by octoshrimpy** [comment]
@gcb thank you for the summary!

my fork quik was forked from the 3.10 branch. 

The difficulty in following the commits is because i'm learning git deployments hands on, and making _a lot_ of mistakes :sweat_smile: 

I've implemented a lot of the PRs that currently are open towards qksms, and am working through bug fixes that are present in the current f-droid QKSMS. 

My goal with quik is to do more than an sms app (like you mentioned, sms isn't great) but instead to have a plugin system that works much like matrix bridges, but locally on your device. no need to have multiple apps, when you can load up the plugins and let one single app do the same job. 


@Retardium is corect, SMT is no longer a viable tool, and fossify is the new alternative to that. Adding to the mess, f-droid also underwent some serious drama recently, with 2/3 of the community board leaving: https://news.ycombinator.com/item?id=38517576

**Comment by KaKi87** [comment]
> Anyone actually uses SMS in this day

Yes : for people who won't use Signal and only offer GAFAMs as alternatives.

> (unsafe) two factor auth?

I avoid these as much as possible.

> google et al are all replacing SMS with RCS

I won't use that either.

**Comment by octoshrimpy** [comment]
also RCS isn't a viable option until we can host our own TURN-like servers, as of right now RCS is a walled garden, as mentioned in https://github.com/octoshrimpy/quik/issues/22

**Comment by octoshrimpy** [comment]
> > Anyone actually uses SMS in this day
> 
> Yes : for people who won't use Signal and only offer GAFAMs as alternatives.

Signal doesn't let us host our own servers, takes a while to update their own servers, and has pushed crypto in-app. I personally no longer trust them. 

https://github.com/dessalines/essays/blob/master/why_not_signal.md

**Comment by KaKi87** [comment]
> My goal with quik is to do more than an sms app (like you mentioned, sms isn't great) but instead to have a plugin system that works much like matrix bridges, but locally on your device. no need to have multiple apps, when you can load up the plugins and let one single app do the same job.

About that, [here](https://github.com/octoshrimpy/quik/commit/9c59374fad2d24cbc8b768a476200330139e3c28#commitcomment-130599086)'s the comment that I made on your README, did you see it ?

> Wow, that's one pretty damn big objective, do you think using an initially SMS-based app would be a good project to start from ?
> 
> Given that it's originally just an SMS-based app, i.e. that it wasn't designed for community-related features such as profiles, channels, threads, etc., I'd tend to believe that it could be too hard to add these in it. 🤔

> Signal doesn't let us host our own servers

Self-hosting Signal is possible, just not federating. It uses phone number verification so that accounts can't be stolen, duplicated or abused.

> has pushed crypto in-app

In order to fund the project.

> I personally no longer trust them.

The server and client remain open source and end-to-end encryption is verifiable.

**Comment by octoshrimpy** [comment]
> About that, [here](https://github.com/octoshrimpy/quik/commit/9c59374fad2d24cbc8b768a476200330139e3c28#commitcomment-130599086)'s the comment that I made on your README, did you see it ?

I totally missed it! replied.

> Self-hosting Signal is possible, just not federating. It uses phone number verification so that accounts can't be stolen, duplicated or abused.

you are correct, but that would be the equivalent of self-hosting reddit server and not federating. that would still cause a single centralized point, easy for attacks, and lock "my" users away from "your" users, causing a systematic split that I'm personally not okay with.
 
> In order to fund the project.

True, all OSS has monetary problems, but for a company paying devs to create new features like stories, wouldn't it be smarter to instead save the money and do bug fixes for the time being instead? I'm no economics expert, but it all sounds weirdy to me. 
 
> The server and client remain open source and end-to-end encryption is verifiable.

that is also correct. However, no federation from server to server, and marlinspike has often [threatened shutting down 3rd party clients](https://github.com/LibreSignal/LibreSignal/issues/37#issuecomment-217211165). I don't see that as a healthy ecosystem to join, as a dev.

**Comment by KaKi87** [comment]
> a systematic split that I'm personally not okay with

Well, that's what Matrix is for, I see that you already use it. ^^

> wouldn't it be smarter to instead save the money and do bug fixes for the time being instead?

Unfortunately no, because features attract new people, or at least make it easier for existing users to convince more people to come instead of forcing them to use big tech platforms.

> marlinspike has often threatened shutting down 3rd party clients

Actually, I read nothing wrong in the comment you linked, which says that they're fine with third-party apps, just not using the Signal server and name, which is logical, because these are also potential sources of abuse.

**Comment by octoshrimpy** [comment]
> Unfortunately no, because features attract new people, or at least make it easier for existing users to convince more people to come instead of forcing them to use big tech platforms.

that may be true. Personally, I'll recommend a rock-solid app, with devs that listen to the users, over one that keeps coming up with half-baked or non-community-voted-for features. I was a signal user, and convinced a lot of people to use it, but then they removed the SMS feature claiming it was too hard to explain to users which messages were encrypted and which were not (UX problem) and then shipped crypto, followed by stories the updates after. I uninstalled at that point.

> Actually, I read nothing wrong in the comment you linked, which says that they're fine with third-party apps, just not using the Signal server and name, which is logical, because these are also potential sources of abuse.

If you have the source code for the server and the client, and host it all yourself, and there is no federation with any other server or any other client, at that point aren't you the 1st party, and the original source is no longer accessible, meaning it's no longer 3rd party server/client?

as for "potential sources of abuse", look at all of activitypub. it's still taking shape, but the communities sort themselves out, and figure out what is good and bad to use. 

I agree RCS is not viable right now. SMS is Very Not Great. for privacy and security, anything not self-hosted, whose servers sit in the US are (personally) out of question.

**Comment by KaKi87** [comment]
> at that point aren't you the 1st party

That's true.

> as for "potential sources of abuse", look at all of activitypub. it's still taking shape, but the communities sort themselves out, and figure out what is good and bad to use.

And I love the Fediverse. But, that only works for public communication, while community-based moderation can't work for private end-to-end encrypted messaging : each time someone spams, each victim has to individually block the sender.

**Comment by octoshrimpy** [comment]
> while community-based moderation can't work for private end-to-end encrypted messaging : each time someone spams, each victim has to individually block the sender.

isn't that what we in the US are experiencing with spam calls as well? Seems like the general consensus solution is something along the lines of "don't even bother telling  the user that there was a call unless it's from a known contact, or user is expecting a call." Not pretty, and puts the solution on the clientside/user. although without identifiers (like signal and phone numbers) this wouldn't be as much of a problem, I would think. again, not sure, just spitballin' ideas with you. :)

**Comment by inson1** [comment]
SMT apps on F-droid are safe :) If ofc F-droid doesnt die
or just dont update it

**Comment by gcb** [comment]
I honestly think you all are missing the point here.

SMS is not great, do not use it. run away. I do. But sometimes you need to receive/send that one annoying message.

The discussion here: User needs to use SMS. Period. Nothing else. Which application to use?

Do not try to adapt a dead sms app to whatever else you want. I mean, by all means to use the code, but fork and just mention "UI was from so and so". You do not need to associated with the original app if you don't want to support the SMS codebase.

**Comment by octoshrimpy** [comment]
> SMS is not great, do not use it. run away. I do. But sometimes you need to receive/send that one annoying message.

In the USA, SMS is still very much used everywhere, one cannot simply run away.

> Do not try to adapt a dead sms app to whatever else you want.

Why not? Is it not my choice, in the end? Personally, I'm using qksms (and quik) as a means to learn kotlin.  

> I mean, by all means to use the code, but fork and just mention "UI was from so and so". You do not need to associated with the original app if you don't want to support the SMS codebase.

But I do want to support the SMS codebase. SMS is very much alive (sadly), and many users need a good updated app, considering all the news lately.

**Comment by Steve-CCity** [comment]
Thanks @octoshrimpy 
I'm definitely not happy that I'm back to using Google message app.  Sure hoping for a Foss alternative.

**Comment by gcb** [comment]
> SMS is very much alive

It is being sunset. What you call SMS today is iMessage(tm) and googleMessage(tm) both which use RCS (with closed extensions) and fallback to sms/mms. But not for long.

My point is that adding features here is wasting time. By all means maintain this, as a simple sms client, or overgrow it to something more complex. Not saying to drop SMS. But stop adding features to this as it is a dead end.

**Comment by KaKi87** [comment]
> What you call SMS today is iMessage(tm) and googleMessage(tm) both which use RCS (with closed extensions) and fallback to sms/mms. But not for long.

I will never use RCS, not even if SMS ends up shut down.

That said, I don't believe that it will be shut down anytime soon, at least in my country, because even PSTN and 2G aren't despite being supposed to for years now.

**Comment by octoshrimpy** [comment]
> What you call SMS today is iMessage(tm) and googleMessage(tm)

Not at all. what I refer to SMS is using the SMS protocol. RCS as it stands is a walled garden being gatekept by Google, who allowed Samsung and Verizon access to their blackbox RCS servers. 

The way you described SMS feels like saying using linux isn't worth it because most computers ship with Windows or macOS.

Like @KaKi87 mentioned, SMS over 2G still operates just fine. 
 
> My point is that adding features here is wasting time. 

Okay. It's my schedule, is it not? By continuing to tell me to Not Do The Thing you're wasting your own time, haha.
 

> or overgrow it to something more complex [...] But stop adding features 

I'm not sure I follow. Are you telling me to continue to add features, or to stop adding features?

> Stop adding features to this as it is a dead end.

100 new stars in a month and half with no advertising or talking about it (except for this one comment thread) would have to disagree with you. 

---

@KaKi87:
> I will never use RCS, not even if SMS ends up shut down

agreed. until I can host it myself, it's not worth anything to me. Granted, RCS is just matrix/xmpp protocol with a fancy name. Wish we could all just agree to stick to what works instead of trying to reinvent the wheel and then try to sell it to someone..

**Comment by gcb** [comment]
I don't know what to say man. you dont control the networks. US doesn't have even 3G anymore. latin america is offsetting 2g (even car trackers IOT sims are 4G now). Europe is holding to 2G for some IOT holdouts, but as soon as the telco move those big clients 2G will go down in the same day.

see https://maya.land/monologues/2023/12/26/laptop-of-the-magi-apple-antitrust.html

**Comment by allanonmage** [comment]
> > > Anyone actually uses SMS in this day
> > 
> > 
> > Yes : for people who won't use Signal and only offer GAFAMs as alternatives.
> 
> Signal doesn't let us host our own servers, takes a while to update their own servers, and has pushed crypto in-app. I personally no longer trust them.
> 
> https://github.com/dessalines/essays/blob/master/why_not_signal.md

The fact that nowhere in his essay does he mention Threema means that he doesn't really know much about encrypted messengers.  Not sure why Threema goes ignored by Silicon Vallyers, but it's often ignored or baselessly rejected for no reason at all.  It's fine if someone doesn't like it or found a problem with it, but to not mention it is lying by omission, which makes me suspect that people have an agenda to ignore it, and/or why I don't think the author knows that much about encrypted messengers.

**Comment by inson1** [comment]
@allanonmage "If I cant read, I cant stop being offended." lol
That essay isnt perfect, but really reading is important, go learn it. He mentioned threema and why he doesnt like it.

**Comment by mddvul22** [comment]
@octoshrimpy how is your fork coming along?  I'm thinking I may have to move back to the stock messaging app that comes with Lineage OS.  Maybe the fork of Simple SMS Messenger might be the way to go.  Looks like their fork just had a new release 18 hours ago.

**Comment by mddvul22** [comment]
Looks like the Fossify Messenger app (fork of Simple SMS Messenger) is now in Fdroid, also.

**Comment by octoshrimpy** [comment]
> @octoshrimpy how is your fork coming along? 

it's coming great! I'm learning a lot. There was an update recently with some bug fixes, and i've got a MR to get it on fdroid waiting to be accepted.
